packages/grant-explorer/src/features/api/voting.ts

```diff
@@ -0,0 +1,214 @@
+        nonce,
+        permitSignature.v,
+        permitSignature.r,
+        permitSignature.s
+      );
+    } else {
+      tx = await mrcImplementation.voteERC20Permit(
+        Object.values(groupedVotes),
+        Object.keys(groupedVotes),
+        Object.values(groupedAmounts),
+        Object.values(groupedAmounts).reduce((acc, b) => acc.add(b)),
+        token.address,
+        deadline,
+        permitSignature.v,
+        permitSignature.r,
+        permitSignature.s
+      );
+    }
+  } else {
+    /* Tried voting using erc-20 but no permit signature provided */
+    throw new Error(
+      "Tried voting using erc-20 but no permit signature provided"
+    );
+  }
+
+  const result = await handleTransaction(tx);
+
+  if (result.error) {
+    // handle error case
+    throw new Error(result.error);
+  } else {
+    console.log("✅ Transaction hash: ", result.txHash);
+
+    return {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      txBlockNumber: result.txBlockNumber!,
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      txHash: result.txHash!,
+    };
+  }
+};
+
+type SignPermitProps = {
+  signer: JsonRpcSigner;
+  contractAddress: string;
+  erc20Name: string;
+  owner: string;
+  spender: string;
+  deadline: number;
+  chainId: number;
+};
+
+type Eip2612Props = SignPermitProps & {
+  value: BigNumber;
+  nonce: BigNumber;
+};
+
+type DaiPermit = SignPermitProps & {
+  nonce: BigNumber;
+};
+
+/* Signs a permit for EIP-2612-compatible ERC-20 tokens */
+export const signPermit2612 = async ({
+  signer,
+  contractAddress,
+  erc20Name,
+  owner,
+  spender,
+  value,
+  deadline,
+  nonce,
+  chainId,
+}: Eip2612Props) => {
+  const types = {
+    Permit: [
+      { name: "owner", type: "address" },
+      { name: "spender", type: "address" },
+      { name: "value", type: "uint256" },
+      { name: "nonce", type: "uint256" },
+      { name: "deadline", type: "uint256" },
+    ],
+  };
+
+  const domainData = {
+    name: erc20Name,
+    version: "2",
+    chainId: chainId,
+    verifyingContract: contractAddress,
+  };
+
+  const message = {
+    owner,
+    spender,
+    value,
+    nonce,
+    deadline,
+  };
+
+  const signature = (await signer._signTypedData(
+    domainData,
+    types,
+    message
+  )) as Hex;
+  const [r, s, v] = [
+    slice(signature, 0, 32),
+    slice(signature, 32, 64),
+    slice(signature, 64, 65),
+  ];
+  return { r, s, v: hexToNumber(v) };
+};
+
+export const signPermitDai = async ({
+  signer,
+  contractAddress,
+  erc20Name,
+  owner,
+  spender,
+  deadline,
+  nonce,
+  chainId,
+}: DaiPermit) => {
+  const types = {
+    Permit: [
+      { name: "holder", type: "address" },
+      { name: "spender", type: "address" },
+      { name: "nonce", type: "uint256" },
+      { name: "expiry", type: "uint256" },
+      { name: "allowed", type: "bool" },
+    ],
+  };
+
+  const domainData = {
+    name: erc20Name,
+    version: "1",
```

Comment:
let's take the version calling the `version` function in the contract

Reply:
As resolved in DMs, the version fn is unfortun. not part of the spec, so hardcoded it in the payout tokens, and default to 1.